helm/templates/_helpers.tpl

```diff
@@ -86,16 +93,20 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
     {{- if eq $validator.type "notaryv1" -}}
         {{- if $validator.auth -}}
             {{- if $validator.auth.secret_name -}}
-                {{- $_ := set $external_secret $validator.name $validator.auth.secret_name -}}
+    - name: {{ $validator.name }}-vol
+  mountPath: /app/connaisseur-config/{{ $validator.name }}
```

Comment:
What happens is `$validator.name == '../connaisseur-config/other-validator`, or some other malicious path? It looks like it would be possible to produce collisions here.

Reply:
First off, I think that would be a path traversal via configuration of Connaisseur against Connaisseur which I would consider an unlikely case and actually would not know how to fix it at this point. However, I tested and there is a regex validation that fails when attempting traversal:

```bash
Error: release connaisseur failed, and has been uninstalled due to atomic being set: Deployment.apps "connaisseur-deployment" is invalid: [spec.template.spec.volumes[6].name: Invalid value: "../default/-vol": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), spec.template.spec.containers[0].volumeMounts[6].name: Not found: "../default/-vol"]
```

Reply:
This is fine under the assumption that the validation of the name doesn't change significantly, e.g. allowing a dot `.` only. I cannot judge if that could happen, but I wouldn't be 100% certain that it doesn't: The error message you quoted mentions DNS-1123 labels, and I fail to see why volume names should be validated like hostnames. So perhaps that's not going to be the case forever.

One solution would be to hash the name before putting it in the path. Let me know what you think; if you feel differently, I won't insist further.

Reply:
I am not a big fan of hashing (and also would not know how to do it in YAML), but also think it is not a critical thing.

Maybe getting a second opinion by @Starkteetje, @phbelitz or @annekebr helps

Reply:
yea ... if you ask me, things could stay as they are, but there are some options:

as peter mentioned we could hash the name (https://helm.sh/docs/chart_template_guide/function_list/#sha256sum). same would need to be done in the connaisseur app code.

alternatively we can replace unwanted characters , like `.` and `/` (https://helm.sh/docs/chart_template_guide/function_list/#replace)

Reply:
think this is the case at several, points in the charts. so let's create an issue and fix it completely at a later time. OK?

Reply:
Sure. But as you said, I'm not even sure if it needs to be fixed / whether it really is a problem. I just wanted to raise my concerns.